runtime/parachains/src/paras_inherent.rs

```diff
@@ -1234,21 +1229,31 @@ mod tests {
 					Pallet::<Test>::create_inherent_inner(&inherent_data.clone()).unwrap();
 				assert!(limit_inherent_data != expected_para_inherent_data);
 
+				// Three disputes is overlength (see previous test), so we expect to only see 2 disputes
 				assert_eq!(limit_inherent_data.disputes.len(), 2);
+				// Ensure disputes are filtered as expected
 				assert_eq!(limit_inherent_data.disputes[0].session, 1);
 				assert_eq!(limit_inherent_data.disputes[1].session, 2);
+				// Ensure all bitfields are included as these are still not overlength
+				assert_eq!(
+					limit_inherent_data.bitfields.len(),
+					expected_para_inherent_data.bitfields.len()
+				);
+				// Ensure that all backed candidates are filtered out as either would make the block overlength
+				assert_eq!(limit_inherent_data.backed_candidates.len(), 0);
 
 				// The schedule is still empty prior to calling `enter`. (`create_inherent_inner` should not
 				// alter storage, but just double checking for sanity).
 				assert_eq!(<scheduler::Pallet<Test>>::scheduled(), vec![]);
 
 				assert_eq!(Pallet::<Test>::on_chain_votes(), None);
-				// Call enter with our 2 backed candidates
+				// Call enter with our 2 disputes
 				assert_ok!(Pallet::<Test>::enter(
 					frame_system::RawOrigin::None.into(),
 					limit_inherent_data,
 				));
 
+				// Ensure that calling enter with 3 disputes and 2 candidates is overlength
```

Comment:
I think this test is not doing what you want it to do - see https://github.com/paritytech/polkadot/pull/4202/commits/58fc1126a4d79bc07202f24b0a47b5c38dda79e5#r743589515

Reply:
You mean because the weight has already been applied based on the above enter function?

Reply:
I think we discussed on call but just to clarify all the storage like AvailabilityCores, Included etc gets mutated by `enter`, so calling it twice in a row will not have the intended affect